src/analysis/normed_space/basic.lean

```diff
@@ -310,12 +314,42 @@ by simp only [inv_eq_one_div, norm_div, norm_one]
            ... = ∥a∥*∥a^k∥ : by rw normed_field.norm_mul
            ... = ∥a∥ ^ (k + 1) : by rw normed_field.norm_pow; simp [pow, monoid.pow]
 
+@[simp] lemma normed_field.norm_fpow {α : Type*} [normed_field α] (a : α) : ∀n : ℤ,
+  ∥a^n∥ = ∥a∥^n
+| (n : ℕ) := normed_field.norm_pow a n
+| -[1+ n] := by simp [fpow_neg_succ_of_nat]
+
 instance : normed_field ℝ :=
 { norm := λ x, abs x,
   dist_eq := assume x y, rfl,
   norm_mul := abs_mul }
 
-lemma real.norm_eq_abs (r : ℝ): norm r = abs r := rfl
+instance : nondiscrete_normed_field ℝ :=
+{ non_trivial := ⟨2, by unfold norm; rw abs_of_nonneg; by norm_num⟩ }
+
+lemma real.norm_eq_abs (r : ℝ) : norm r = abs r := rfl
+
+instance : normed_field ℂ :=
+{ norm := λ x, x.abs,
+  dist_eq := assume x y, rfl,
+  norm_mul := complex.abs_mul }
+
+instance : nondiscrete_normed_field ℂ :=
+{ non_trivial := ⟨2, by simp [norm]; norm_num⟩ }
+
+lemma complex.norm_eq_abs (r : ℂ) : norm r = r.abs := rfl
+
+lemma exists_norm_lt_one [i : nondiscrete_normed_field α] : ∃x : α, 0 < ∥x∥ ∧ ∥x∥ < 1 :=
```

Comment:
You don't need to name this instance `i`. You can write `rcases nondiscrete_normed_field.non_trivial α with ⟨y, hy⟩` where you used `i`. If you don't like how long it is you can write a lemma restating this axiom with another name.